PLT-5156 Fix Channel Switcher Console Errors.

grundleborg opened the pull request:
#### Summary
Fix console errors and error reporting in UI for Channel Switcher. Addresses both parts of PLT-5156.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5156

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file updates

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-0d60ee7d24f47cf73.spinmint.com:8065/pr5058

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-0d60ee7d24f47cf73

jasonblais commented:
Hey @grundleborg tested it out:

#### 1: 

Repro:
1) In the above test server, hit CTRL+K to open channel switcher
2) Type "sa"  --> observed: no results
![image](https://cloud.githubusercontent.com/assets/13119842/21959216/60ebd3fc-da8e-11e6-9e31-e949a2d2f4a4.png)

3) Hit Enter or press "Switch"

Observed: Modal closes and view switches to channel `zpfrfegmotafethsbz` with handle `jxawnwekrbemfdzd` [there is no `sa` part of the channel name or handle]
![image](https://cloud.githubusercontent.com/assets/13119842/21959219/6bf30b62-da8e-11e6-943b-408b761c9315.png)
Expected: Channel switcher doesn't close and "No matches found" message appears below the search input box

#### 2:

I'm not sure how you implemented the [first part of the ticket](https://mattermost.atlassian.net/browse/PLT-5156) when a user hasn't entered anything to the input box, but let's keep the expected as specified in the ticket [no action].

It was discussed in UX and having feedback to the end user when nothing is pressed didn't feel necessary and wouldn't be consistent with other similar scenarios (like hitting Enter in an empty message box)

grundleborg commented:
@jasonblais 

1) This appears to have been an unrelated bug with the channel switcher that caused it to retain a "selected channel" even after the user had changed the entered text. Fixed it anyway.

2) Fixed.

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-0b7aa4692ac4ff5fa.spinmint.com:8065/pr5058

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-0b7aa4692ac4ff5fa

mattermod commented:
Spinmint test server destroyed

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-0e08e295442c07ee8.spinmint.com:8065/pr5058

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-0e08e295442c07ee8

mattermod commented:
Spinmint test server destroyed

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-0aefc8d6950f2adaa.spinmint.com:8065/pr5058

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-0aefc8d6950f2adaa

mattermod commented:
Spinmint test server destroyed
